x-pack/plugins/remote_clusters/__jest__/client_integration/list/remote_clusters_list.test.js

```diff
@@ -103,9 +103,10 @@ describe('<RemoteClusterList />', () => {
       expect(tableCellsValues.length).toBe(10);
     });
 
-    // Skipped until we can figure out how to get this test to work.
-    test.skip('search works', () => {
-      form.setInputValue(find('remoteClusterSearch'), 'unique');
+    test('search works', () => {
+      const input = find('remoteClusterSearch');
+      input.instance().value = 'unique';
+      input.simulate('keyup', { key: 'Enter', keyCode: 13, which: 13 });
```

Comment:
Hmm, so it seems like the consumer needs to know a bit about the `setInputValue` implementation in order to use it correctly. Are there any changes we can make to the testbed interface to reduce/remove this burden? For example, what if we added this function to testbed:

```js
// General idea, needs TS typings and maybe support for `isAsync` for parity with `setInputValue`
const submitInputValue = (input, value) => {
  const input = find(input);
  input.instance().value = value;
  input.simulate('keyup', { key: 'Enter', keyCode: 13, which: 13 });
};
```

Would having `setInputValue` and `submitInputValue` available on the testbed interface be enough for consumers to infer what the difference is between the two? If not, is there a better solution?

Reply:
Before going that path I think we should understand why this input reacts to a `keyup` event and not an `onChange`. What is the component being rendered as input?

Instead of adding more method to the testBed I would suggest to mock that input component as we do with the ComboBox. After that we can setValue normally with (`form.setInputValue()`)

```js
jest.mock('@elastic/eui', () => {
  const original = jest.requireActual('@elastic/eui');

  return {
    ...original,
    EuiComboBox: (props: any) => (
      <input
        data-test-subj={props['data-test-subj'] || 'mockComboBox'}
        onChange={async (syntheticEvent: any) => {
          props.onChange([syntheticEvent['0']]);
        }}
      />
    ),
  };
});
```

Reply:
Thanks a lot for your review, @cjcenizal and @sebelga! 
I found out that the `keyup` event is specific to `EuiSearchBar` ([EUI test](https://github.com/elastic/eui/blob/e6fd955eb073a9455932e8b761424b4e015bd444/src/components/search_bar/search_bar.test.tsx#L108)) that is used in the table, so I thought adding a helper method as CJ suggested should work, similar to want we do with `EuiCombobox` ([setComboboxValue](https://github.com/elastic/kibana/blob/8de766904d28da27ad707a920ade9510e86a2941/packages/kbn-test/src/jest/utils/testbed/testbed.ts#L250)) 

I also unskipped 2 tests in CCR after fixing some `undefined` errors being thrown in the tests.

Reply:
Just wanted to give a bit more context of my suggestion above.
 
I am the one who originally wrote the `setComboboxValue`. But in practice (in most of our tests), as @alisonelizabeth has shown me, we mock the EuiComboBox component and convert it to a simple `<input />`. At that stage, it is just like any other input and we don't need to do anything special in testBed (we can simply call `form.setInputValue()`.  
After seeing it in practice I think that is a better approach than having a handler like `setComboboxValue` and adding more handlers (`setSearchBarValue`) because if EUI changes anything in their implementation it could affect our tests and thus break CI for people.

It also lower the API surface of the testBed (remember the power of subtraction 😊 )

If you look at the [setup_environment.tsx](https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_management/public/application/components/mappings_editor/__jest__/client_integration/helpers/setup_environment.tsx) from the mappings editor you can see how we mock 3 EUI inputs and convert them to simple inputs to interact with.

I could see  L17-L55 from that file be extended further more to include the search bar component. And then we could export that file in a single place (from the "es_ui_shared" folder) to mock all the EUI components which can not easily been interacted with in tests.

With that in place we simply need to import that file at the top of all our tests file and we'd be good.

Reply:
Thanks for your explanation, Seb! I see where you're coming from. I do like the approach you're describing better. As you mentioned, it avoids testing EUI, which I think is the right thing to do because EUI already tests all of its own components. @yuliacech WDYT?

Reply:
Thanks for this discussion, @sebelga and @cjcenizal! 
I agree with Seb about subtracting and not adding more to the existing `testbed` interface, especially such implementation specific things like search box reacting to the `keyup` event. So I followed the suggestion to use mocks instead. In this PR I have 2 separate mocks for remote clusters and ccr tests and I'd like to open a discuss issue to find the best way/place where EUI mocks could live. In a follow up PR we could then remove duplicated mocks from our tests, import a common mock file and maybe even delete unused `setComboboxValue` from the testbed interface.

Reply:
Thanks @yuliacech . Yes we do have mocked declared in many places. And not all of them are exactly the same (most are though). So yes it would need a bit of cleaning 😊